packages/tools/examples/labelmapSegmentationDynamicThreshold/index.ts

```diff
@@ -0,0 +1,525 @@
+element2.style.height = size;
+element3.style.width = size;
+element3.style.height = size;
+
+// Disable right click context menu so we can have right click tools
+element1.oncontextmenu = (e) => e.preventDefault();
+element2.oncontextmenu = (e) => e.preventDefault();
+element3.oncontextmenu = (e) => e.preventDefault();
+
+viewportGrid.appendChild(element1);
+viewportGrid.appendChild(element2);
+viewportGrid.appendChild(element3);
+
+content.appendChild(viewportGrid);
+
+const instructions = document.createElement('p');
+instructions.innerText = `
+  Hover - show preview of segmentation tool
+  Left drag to extend preview
+  Left Click (or enter) to accept preview
+  Reject preview by button (or esc)
+  Hover outside of region to reset to hovered over segment index
+  Shift Left - zoom, Ctrl Left - Pan, Alt Left - Stack Scroll
+  `;
+
+content.append(instructions);
+
+const brushInstanceNames = {
+  ThresholdCircle: 'ThresholdCircle',
+  CircularBrush: 'CircularBrush',
+  CircularEraser: 'CircularEraser',
+  SphereBrush: 'SphereBrush',
+  SphereEraser: 'SphereEraser',
+  ScissorsEraser: 'ScissorsEraser',
+};
+
+const brushStrategies = {
+  [brushInstanceNames.CircularBrush]: 'FILL_INSIDE_CIRCLE',
+  [brushInstanceNames.CircularEraser]: 'ERASE_INSIDE_CIRCLE',
+  [brushInstanceNames.SphereBrush]: 'FILL_INSIDE_SPHERE',
+  [brushInstanceNames.SphereEraser]: 'ERASE_INSIDE_SPHERE',
+  [brushInstanceNames.ThresholdCircle]: 'THRESHOLD_INSIDE_CIRCLE',
+  [brushInstanceNames.ScissorsEraser]: 'ERASE_INSIDE',
+};
+
+const brushValues = [
+  brushInstanceNames.ThresholdCircle,
+  brushInstanceNames.CircularBrush,
+  brushInstanceNames.CircularEraser,
+  brushInstanceNames.SphereBrush,
+  brushInstanceNames.SphereEraser,
+];
+
+const optionsValues = [
+  ...brushValues,
+  RectangleScissorsTool.toolName,
+  CircleScissorsTool.toolName,
+  SphereScissorsTool.toolName,
+  brushInstanceNames.ScissorsEraser,
+  PaintFillTool.toolName,
+];
+
+const previewColors = {
+  0: [255, 255, 255, 128],
+  1: [0, 255, 255, 255],
+};
+
+// ============================= //
+addDropdownToToolbar({
+  options: { values: optionsValues, defaultValue: BrushTool.toolName },
+  onSelectedValueChange: (nameAsStringOrNumber) => {
+    const name = String(nameAsStringOrNumber);
+    const toolGroup = ToolGroupManager.getToolGroup(toolGroupId);
+
+    // Set the currently active tool disabled
+    const toolName = toolGroup.getActivePrimaryMouseButtonTool();
+
+    if (toolName) {
+      toolGroup.setToolDisabled(toolName);
+    }
+
+    if (brushValues.includes(name)) {
+      toolGroup.setToolActive(name, {
+        bindings: [{ mouseButton: MouseBindings.Primary }],
+      });
+    } else {
+      const toolName = name;
+
+      toolGroup.setToolActive(toolName, {
+        bindings: [{ mouseButton: MouseBindings.Primary }],
+      });
+    }
+  },
+});
+
+const thresholdOptions = new Map<string, any>();
+thresholdOptions.set('Dynamic r=0', { isDynamic: true, delta: 0 });
```

Comment:
what does r=0 mean? I see it is setting the delta to 0, is this the radius or the delta for threshold? in either case we better be consistent